homeassistant/components/toon/config_flow.py

```diff
@@ -0,0 +1,156 @@
+"""Config flow to configure the Toon component."""
+import logging
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import callback
+
+from .const import (
+    CONF_CLIENT_ID, CONF_CLIENT_SECRET, CONF_DISPLAY, CONF_TENANT,
+    DATA_TOON_CONFIG, DEFAULT_TENANT, DOMAIN)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@callback
+def configured_displays(hass):
+    """Return a set of configured Toon displays."""
+    return set(
+        entry.data[CONF_DISPLAY]
+        for entry in hass.config_entries.async_entries(DOMAIN)
+    )
+
+
+@config_entries.HANDLERS.register(DOMAIN)
+class ToonFlowHandler(config_entries.ConfigFlow):
+    """Handle a Toon config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    def __init__(self):
+        """Initialize the Toon flow."""
+        self.displays = None
+        self.username = None
+        self.password = None
+        self.tenant = None
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initiated by the user."""
+        app = self.hass.data.get(DATA_TOON_CONFIG, {})
+
+        if not app:
+            return self.async_abort(reason='no_app')
+
+        return await self.async_step_authenticate(user_input)
+
+    async def _show_authenticaticate_form(self, errors=None):
+        """Show the authentication form to the user."""
+        data_schema = vol.Schema({
+            vol.Required(CONF_USERNAME): str,
+            vol.Required(CONF_PASSWORD): str,
+            vol.Optional(CONF_TENANT, default=DEFAULT_TENANT): str,
+        })
```

Comment:
Just FYI, the order of these items can be messed up because the min. python version HA supports does not have a specific requirement that dicts are ordered. If that is a problem, consider using `OrderedDict()`

Reply:
Good catch! 

https://github.com/home-assistant/home-assistant/blob/ac502980a214d1702c1b486e8b98ce09dea19209/homeassistant/components/esphome/config_flow.py#L42-L44

Reply:
Thanks, @OttoWinter, this has been addressed. Nice catch indeed!